.github/workflows/financebench.yml

```diff
@@ -0,0 +1,146 @@
+
+      - name: Set up Rust
+        uses: ./.github/actions/setup-rust
+        with:
+          os: 'darwin'
+
+      - name: Build the testoperator
+        run: |
+          cargo build -p testoperator --release
+
+      - name: make testoperator executable
+        run: |
+          mv target/release/testoperator testoperator
+          chmod +x testoperator
+          
+      - name: Save spice artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: testoperator
+          path: testoperator
+
+  build-spiced:
+    name: Build spiced
+    runs-on: 'spiceai-macos'
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Rust
+        uses: ./.github/actions/setup-rust
+        with:
+          os: darwin
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v3
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build spiced
+        if: matrix.target.target_os != 'darwin'
+        run: |
+          make -C bin/spiced SPICED_NON_DEFAULT_FEATURES="models"
+
+      - name: make spiced executable
+        if: matrix.target.target_os != 'windows'
+        run: |
+          mv target/release/spiced spiced
+          chmod +x spiced
+
+      - name: Save spice artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: spiced
+          path: spiced
+
+  setup-matrix:
+    name: Setup matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.setup-matrix.outputs.result }}
+
+    steps:
+      - name: Set up matrix
+        uses: actions/github-script@v7
+        id: setup-matrix
+        with:
+          script: |
+            const matrix = [
+              {
+                name: "gpt-4o",
+                spicepod: "./test/financebench/spicepod_gpt-4o.yaml"
+              }
+            ];
+
+            const model_option = context.payload.inputs.model_option;
+            let filtered = matrix;
+
+            if (model_option !== 'all') {
+              filtered = filtered.filter(m => m.name === model_option);
+            }
+
+            return filtered;
+
+  run-testoperator:
+    name: financebench benchmark
+    needs:
+      - build-testoperator
+      - build-spiced
+      - setup-matrix
+    runs-on: 'spiceai-macos'
+    strategy:
+      fail-fast: false
+      matrix:
+        target: ${{ fromJson(needs.setup-matrix.outputs.matrix) }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Download testoperator
+        uses: actions/download-artifact@v4
+        with:
+          name: testoperator
+  
+      - name: Download spiced
+        uses: actions/download-artifact@v4
+        with:
+          name: spiced
+
+      - name: Mark binaries executable
+        run: |
+          chmod +x ./testoperator
+          chmod +x ./spiced
+
+      - name: Run FinanceBench evals
+        env:
+          SPICEAI_FINANCEBENCH_API_KEY: ${{ secrets.SPICE_SECRET_SPICEAI_FINANCEBENCH_API_KEY}}
```

Comment:
The SPICE_SECRET_SPICEAI_FINANCEBENCH_API_KEY needs to be added to repo secret after PR is merged